agentconnection.cpp

```diff
@@ -1,8 +1,93 @@
 #include "agentconnection.h"
 #include "message.h"
+#include "allocations.h"
 
+#include <QtEndian>
 #include <QTimer>
 
+Entity::Entity(int ronum, const char *data, int length) : RoutingObject(ronum, data, length)
+{
+    QByteArray sk;
+    if (ronum == bwpo::num::ROEntityWKey)
+    {
+        this->offset = 32;
+        sk = QByteArray::fromRawData(data, this->offset);
+        ronum = bwpo::num::ROEntity;
+    }
+
+    Q_ASSERT(ronum == bwpo::num::ROEntity);
+
+    const char* content = this->content();
+    this->vk = QByteArray::fromRawData(data, 32);
+
+    int idx = 32;
+    int ln;
+
+    forever
+    {
+        switch (content[idx])
+        {
+        case 0x02: // Creation date
+            Q_ASSERT_X(content[idx + 1] == 8, "Entity", "invalid creation date");
```

Comment:
it's probably ok to do this for now, but this should not be an assert. The go uses panic() but it is recovered and converted into a normal error higher up in the call chain. In the c++ bindings you need to return an error not crash the program with an assert. Also, the go bindings rely on the fact that out of range accesses to arrays cause a recoverable panic, you may need to double check your logic if you copied it directly from the go because the C++ doesn't give you that.